### PR TITLE
Add huoshan (volcengine/bytedance).

### PR DESCRIPTION
**Pull request by MitsuhaMiyamizu**
It's been a long journey toward this commit, a lot of debug and exploration of their API (which is more standardized compared to the others). The authentication of volcengine is more like Amazon Web Services (AWS) S3.

For more information regarding the implementation of authentication, see: 
https://datatracker.ietf.org/meeting/105/materials/slides-105-oauth-sessb-http-request-signing-in-amazon-web-services-00

tl;dr: Server side will verify the request body, generate an authentication string, and compare it with the one your HTTP request header contains.

### CODE REVIEW

**Comment by GuangchuangYu** [comment]
As you are now using `httr2` in this PR, I would like to invite you to help migrate to `httr2`, <https://github.com/YuLab-SMU/fanyi/issues/7>.

**Comment by GuangchuangYu** [comment]
I run the `volcengine_translate('hello')` and it throws an error.  Can you double-check?

**Comment by MitsuhaMiyamizu** [comment]
> I run the `volcengine_translate('hello')` and it throws an error. Can you double-check?

I checked using ```.volcengine_translate_query()```, and it works, what exactly do you have in the error message.

You can extract the ```.volcengine_translate_query()```, and edit appid and secret manually, see if it works.

**Comment by GuangchuangYu** [comment]
```
> resp
$ResponseMetadata
$ResponseMetadata$RequestId
[1] "202312260912306507E9BF866BA8A4DE31"

$ResponseMetadata$Action
[1] "TranslateText"

$ResponseMetadata$Version
[1] "2020-06-01"

$ResponseMetadata$Service
[1] "translate"

$ResponseMetadata$Region
[1] "cn-north-1"

$ResponseMetadata$Error
$ResponseMetadata$Error$Code
[1] "-403"

$ResponseMetadata$Error$Message
[1] "account status abnormal: unsynchronized"



$ResponseMetaData
$ResponseMetaData$RequestId
[1] "202312260912306507E9BF866BA8A4DE31"

$ResponseMetaData$Action
[1] "TranslateText"

$ResponseMetaData$Version
[1] "2020-06-01"

$ResponseMetaData$Service
[1] "translate"

$ResponseMetaData$Region
[1] "cn-north-1"

$ResponseMetaData$Error
$ResponseMetaData$Error$Code
[1] "-403"

$ResponseMetaData$Error$Message
[1] "account status abnormal: unsynchronized"
```

>$ResponseMetaData$Error$Message
>[1] "account status abnormal: unsynchronized"

I can confirm that I have entered the correct `appid` and `key`.

**Comment by GuangchuangYu** [comment]
It seems accessing via api is not for individuals.

**Comment by MitsuhaMiyamizu** [comment]
Hi, this might be attributed to inadequate permissions. Add the following permissions if you may and try again see if it works.
图片

**Comment by GuangchuangYu** [comment]
still not working after enabling the same permissions.

**Comment by MitsuhaMiyamizu** [comment]
Cannot reproduce here, I installed latest fanyi using ```yulab.utils::install_zip_gh("YuLab-SMU/fanyi")``` and it works well. 
> account status abnormal: unsynchronized

Have you synchronized the system time and checked it, make sure it up-to-date as the authentication operation is time-sensitive? Could you please send the result for doing this:

```sudo timedatectl timesync-status```

img

**Comment by GuangchuangYu** [comment]
sorry, I forgot to enable 'Machine Translation' service.

**Comment by GuangchuangYu** [comment]
It works now

**Comment by MitsuhaMiyamizu** [comment]
Hooray, that's nice :-)
